File: app/src/main/java/de/schooltec/datapass/BootReceiver.java
package de.schooltec.datapass;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

/**
 * Starts the service for receiving connection updates.
 *
 * @author Martin Hellwig
 */
public class BootReceiver extends BroadcastReceiver
{
    @Override
    public void onReceive(Context context, Intent intent)
    {
        ConnectionChangeReceiver.registerReceiver(context);
    }
}

File: app/src/main/java/de/schooltec/datapass/BootReceiver.kt
package de.schooltec.datapass

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

/**
 * Starts the service for receiving connection updates.
 *
 * @author Martin Hellwig
 */
class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action ?: return
        if (action == Intent.ACTION_BOOT_COMPLETED) {
            ConnectionChangeReceiver.registerReceiver(context)
        }
    }
}
